### PR TITLE
Release box after hatching in EggAutonomous

### PR DESCRIPTION
**Pull request by chicknbeer**
I copied MassRelease::release_box to BoxRoutines and used it in EggAutonomous

### CODE REVIEW

File: SerialPrograms/Source/PokemonSV/Programs/Eggs/PokemonSV_EggAutonomous.cpp [hunk 1]
```diff
@@ -341,16 +342,20 @@ void EggAutonomous::hatch_eggs_full_routine(SingleSwitchProgramEnvironment& env,
...
         } // end for each hatched pokemon in party
 
+        // move party back into the box
+        env.log("Unload party");
+        env.console.overlay().add_log("Unload party", COLOR_WHITE);
+        unload_one_column_from_party(env.program_info(), env.console, context, next_egg_column-1, HAS_CLONE_RIDE_POKEMON);
+
         // Get the next egg column
         SomethingInBoxSlotDetector sth_in_box_detector(COLOR_RED);
-        for (; next_egg_column < 6; next_egg_column++){
```
Comment (chicknbeer) [inline SerialPrograms/Source/PokemonSV/Programs/Eggs/PokemonSV_EggAutonomous.cpp:346]: next_egg_column is incremented on line 417 at the end egg batch loop

File: SerialPrograms/Source/PokemonSV/Programs/Eggs/PokemonSV_EggAutonomous.cpp [hunk 2]
```diff
@@ -324,7 +324,8 @@ void EggAutonomous::hatch_eggs_full_routine(SingleSwitchProgramEnvironment& env,
     // The loop to hatch batches of eggs.
     // Each batch consists of at most five eggs.
     // There are at most six batches of eggs in a box.
-    uint8_t next_egg_column = 0; // next egg column in box
+    uint8_t next_egg_column = 1; // next egg column in box
+    // starting at 1 because picnic_party_to_hatch_party already loaded and hatched the first batch.
```
Comment (chicknbeer) [inline SerialPrograms/Source/PokemonSV/Programs/Eggs/PokemonSV_EggAutonomous.cpp:328]: A mistake here. picnic_party_to_hatch_party only load the first column into the party.
The first column is not hatched yet. It is hatched during the first iteration at line 336 hatch_eggs_at_zero_gate

File: SerialPrograms/Source/PokemonSV/Programs/Eggs/PokemonSV_EggAutonomous.cpp [hunk 3]
```diff
@@ -478,8 +493,6 @@ void EggAutonomous::process_one_baby(SingleSwitchProgramEnvironment& env, BotBas
 
         case EggHatchAction::Release:
```
Comment (pifopi) [inline SerialPrograms/Source/PokemonSV/Programs/Eggs/PokemonSV_EggAutonomous.cpp:494]: On phone so hard to check but is this enum value used anywhere else?

**Comment by pifopi** [comment]
Why is it faster to do so ? There is one more step (moving pokemon to the boxes)

File: SerialPrograms/Source/PokemonSV/Programs/Eggs/PokemonSV_EggAutonomous.cpp (hunk 3 above)
Comment (chicknbeer) [reply SerialPrograms/Source/PokemonSV/Programs/Eggs/PokemonSV_EggAutonomous.cpp:494]: I don't think so. It is the default action set on line 436.

**Comment by chicknbeer** [comment]
> Why is it faster to do so ? There is one more step (moving pokemon to the boxes)

It is faster when you expect to release more often than keeping because the program can restart the game before releasing any pokemon at all. For example breeding only for shiny with the shiny charm, you can expect to one in 512 which means 17 boxes of release would be skipped by skipping straight to restarting the game.

**Pull request merged**